File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDODownloadLanguageModel.kt
//
//  IDODownloadLanguageModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///

//private object GsonTester {
//    @JvmStatic
//    fun main(args: Array<String>) {
//        val builder = GsonBuilder()
//        val gson: Gson = builder.create()
//        var jsonString = "{\"use_lang\":1,\"default_lang\":1,\"fixed_lang\":1,\"max_storage_lang\":1,\"lang_array\":[{\"type\":1},{\"type\":2},{\"type\":3},{\"type\":4}]}"
//        val student: IDODownloadLanguageModel =
//            gson.fromJson(jsonString, IDODownloadLanguageModel::class.java)
//        println(student)
//        jsonString = gson.toJson(student)
//        println(jsonString)
//    }
//}

open class IDODownloadLanguageModel(
    useLang: Int,
    defaultLang: Int,
    fixedLang: Int,
    maxStorageLang: Int,
    langArray: List<IDODownloadLanguageType>
) : IDOBaseModel {

    /// Current used language
    @SerializedName("use_lang")
    var useLang: Int = useLang

    /// Default language
    @SerializedName("default_lang")
    var defaultLang: Int = defaultLang

    /// Number of fixed stored languages
    @SerializedName("fixed_lang")
    var fixedLang: Int = fixedLang

    /// Maximum stored languages
    @SerializedName("max_storage_lang")
    var maxStorageLang: Int = maxStorageLang

    /// List of stored language values
    @SerializedName("lang_array")
    var langArray: List<IDODownloadLanguageType> = langArray


    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}

// MARK: - IDOLangArray
open class IDODownloadLanguageType(type: Int) {
    /// Stored language values, ended by 0
    @SerializedName("type")
    var type: Int = type
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDODownloadLanguageModel.swift
//
//  IDODownloadLanguageModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/18.
//

import Foundation

@objcMembers
public class IDODownloadLanguageModel: NSObject, IDOBaseModel {
    /// Current used language
    public var useLang: Int
    /// Default language
    public var defaultLang: Int
    /// Number of fixed stored languages
    public var fixedLang: Int
    /// Maximum stored languages
    public var maxStorageLang: Int
    /// List of stored language values
    public var langArray: [IDODownloadLanguageType]

    enum CodingKeys: String, CodingKey {
        case useLang = "use_lang"
        case defaultLang = "default_lang"
        case fixedLang = "fixed_lang"
        case maxStorageLang = "max_storage_lang"
        case langArray = "lang_array"
    }

    public init(useLang: Int, defaultLang: Int, fixedLang: Int, maxStorageLang: Int, langArray: [IDODownloadLanguageType]) {
        self.useLang = useLang
        self.defaultLang = defaultLang
        self.fixedLang = fixedLang
        self.maxStorageLang = maxStorageLang
        self.langArray = langArray
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

// MARK: - IDOLangArray
@objcMembers
public class IDODownloadLanguageType: NSObject, Codable {
    /// Stored language values, ended by 0
    public var type: Int

    enum CodingKeys: String, CodingKey {
        case type = "type"
    }

    public init(type: Int) {
        self.type = type
    }
}
